### PR TITLE
3412 patient on doc

### PR DESCRIPTION
**Pull request by SCdF**
Webapp version of https://github.com/medic/medic-sentinel/pull/179

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

### CODE REVIEW

**Comment by SCdF** [comment]
@garethbowen I've made the changes you talked about. You don't need to re-review the whole PR:
 - Here are your requested changes: https://github.com/medic/medic-webapp/pull/4005/commits/f1f1bbf6bcdbf50d3d4b2373dd52c93e7f65dc73
 - Here is a weird bug I found and fixed: https://github.com/medic/medic-webapp/pull/4005/commits/a04fcad55669db6fa69636a11c5a46517507cea5

I don't really get why the second one is a bug, but it is, and re-writing the code as I did causes the test I wrote to pass, and stops sentinel from erroring on `-info` docs.

File: sentinel/lib/lineage.js [hunk 1]
```diff
@@ -47,31 +52,55 @@ const minifyContact = contact => {
...
 };
 
-const fetchHydratedDoc = id => {
-  return new Promise((resolve, reject) => {
+const patientLineageByShortcode = shortcode =>
+  new Promise((resolve, reject) => utils.getPatientContactUuid(db, shortcode, (err, uuid) => {
+    if (err) {
+      reject(err);
+    } else {
+      lineageById(uuid).then(resolve);
```
Comment (garethbowen) [inline sentinel/lib/lineage.js]: I think you still have a problem here if `lineageById` fails the error will get lost. I think you either need to catch and reject or find some other way to pass the error out.

File: sentinel/lib/messages.js [hunk 2]
```diff
@@ -175,12 +176,21 @@ module.exports = {
...
+        } else if (doc[recipient]) {
+            // Or directly on the doc
+            phone = doc[recipient];
+        } else if (recipient.indexOf('.') > -1) {
+            // Or multiple layers by executing it as a statement
+            try {
+                phone = objectPath.get(doc, recipient);
+            } catch (err) {
+                logger.error(`Recipient expression "${recipient}" failed on ${doc._id}`);
+            }
```
Comment (garethbowen) [inline sentinel/lib/messages.js]: I don't think objectpath ever throws (although you should confirm) so we don't need the try/catch any more.

Comment (SCdF) [reply sentinel/lib/messages.js]: In classic new-age documentation style it never mentions edge cases anywhere, so I can't be sure without walking through the code. However, I can't *make* it crash, so I guess that's good enough?

**Pull request merged**